CMSgov/design-system
# [WNMGDS-1180] mgov icons

## Summary
Updating mgov icons to follow pattern of icons in core

### Added
1. A `RoundedStarIcon` component since mgov star icons are different than core.
2. A `HamburgerIcon` to core that can be used in mgov

### Changed
Moved the following Icons into an Icon directory for Mgov. These components will eventually be removed:
* Caret Icon
* Checkmark Icon
* Close Icon
* Hamburger Icon
* No Icon

### Deprecated
Marked the following components as deprecated by adding a `console.error` in the component as well as an alert on the documentation page
* Caret Icon
* Checkmark Icon
* Close Icon
* Hamburger Icon
* No Icon

### Removed
All files except the `.svg` file for the following directories in mgov
* Caret
* Checkmark
* Close
* Hamburger
* No

## How to test
I intentionally have not updated the HTML examples for the icons I updated because it's kind of an easy way to see if the icons are the same! (I can do that in a separate PR)

Run the medicare DS locally by running `yarn build`, `yarn start:medicare`

Look at icon pages where I updated the icon to use the core icon version. Note the deprecation warnings:
* [Caret](http://localhost:3000/components/caret-icon/)
* [Checkmark](http://localhost:3000/components/checkmark-icon/)
* [Close](http://localhost:3000/components/close-icon/)
* [Hamburger](http://localhost:3000/components/hamburger-icon/)
* [No](http://localhost:3000/components/No/)

Compare the `Stars` component to the live site and ensure they are the same: [demo branch](http://localhost:3000/components/stars-icons/) vs [live site](https://cmsgov.github.io/design-system/medicare/components/stars-icons/)

Look at the [Icons page](http://localhost:3000/components/icon/) and note the hamburger menu has been added to the list of core icons & the mgov-specific icons have their own section

Note: This PR does not include moving svgs that are "logos".
